### ISSUE TITLE
JRuby enum issue: wrong number of arguments (1 for 2)

### ISSUE THREAD

**Issue by pierre** [killbill/killbill-plugin-framework-ruby#7]
You can reproduce it with the following snippet:

```
    enum_value = 'PROCESSED'

    nb_threads = 2000
    nb_times   = 1000

    threads    = (0..nb_threads).map do |i|
      Thread.new do
        nb_times.times do
          enum = Java::org.killbill.billing.payment.plugin.api.PaymentPluginStatus

          begin
            enum.value_of(enum_value).to_s.should == enum_value
          rescue => e
            puts enum.inspect
            raise e
          end
        end
      end
    end

    threads.each { |thread| thread.join }
```

Broke as early as `i == 8` in my testing.

FYI, `value_of` works fine again when done in the `rescue` block.

**Comment by kares** [comment]
confirmed that this no longer happens using a custom JRuby (1.7.20-SNAPSHOT) built ... all seems stable

**Comment by pierre** [comment]
Sweet! :beers:

**Comment by pierre** [comment]
Fixed in Kill Bill 0.14.0.
